src/lib/syncDocsPath.ts

```diff
@@ -27,10 +30,10 @@ async function pushDoc(
   key: string,
   selectedVersion: string | undefined,
   dryRun: boolean,
-  filePath: string,
+  fileData: ReadDocMetadata,
   type: CommandCategories,
 ) {
-  const { content, data, hash, slug } = readDoc(filePath);
+  const { content, data, filePath, hash, slug } = fileData;
```

Comment:
this refactor (i.e., decoupling the `readDoc` so we can read the contents prior to sorting) is very clever!